dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java

```diff
@@ -0,0 +1,44 @@
+package com.dotcms.util;
+
+import com.dotcms.tika.TikaUtils;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Mime Type Utils
+ * @author jsanca
+ */
+public class MimeTypeUtils {
+
+    /**
+     * Gets the mime type of a file.
+     * @param binary {@link File}
+     * @return String
+     */
+    public static String getMimeType (final File binary) {
+
+        final Path path = binary.toPath();
+        String mimeType = Sneaky.sneak(() -> Files.probeContentType(path));
+
+        if  (!UtilMethods.isSet(mimeType)) {
+
+            mimeType    = Config.CONTEXT.getMimeType(binary.getAbsolutePath());
+
+            if( !UtilMethods.isSet(mimeType)){
+                try {
+                    mimeType = new TikaUtils().detect(binary);
+                } catch(Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)